unix_integration/resolver/src/idprovider/system.rs

```diff
@@ -147,65 +147,58 @@ impl SystemProvider {
+
+        let s_iter = shadow.into_iter().filter_map(|shadow_entry| {
+            let EtcShadow {
+                name,
+                password,
+                epoch_change_days,
+                days_min_password_age,
+                days_max_password_age,
+                days_warning_period,
+                days_inactivity_period,
+                epoch_expire_date,
+                flag_reserved: _,
+            } = shadow_entry;
+
+            if password.is_valid() {
+                let aging_policy = epoch_change_days.map(|change_days| {
+                    AgingPolicy::new(
+                        change_days,
+                        days_min_password_age,
+                        days_max_password_age,
+                        days_warning_period,
+                        days_inactivity_period,
+                    )
+                });
+
+                let expiration_date = epoch_expire_date
+                    .map(|expire| OffsetDateTime::UNIX_EPOCH + time::Duration::days(expire));
 
-        if let Some(shadow) = shadow {
-            let s_iter = shadow.into_iter().filter_map(|shadow_entry| {
-                let EtcShadow {
+                Some((
                     name,
-                    password,
-                    epoch_change_days,
-                    days_min_password_age,
-                    days_max_password_age,
-                    days_warning_period,
-                    days_inactivity_period,
-                    epoch_expire_date,
-                    flag_reserved: _,
-                } = shadow_entry;
-
-                if password.is_valid() {
-                    let aging_policy = epoch_change_days.map(|change_days| {
-                        AgingPolicy::new(
-                            change_days,
-                            days_min_password_age,
-                            days_max_password_age,
-                            days_warning_period,
-                            days_inactivity_period,
-                        )
-                    });
-
-                    let expiration_date = epoch_expire_date
-                        .map(|expire| OffsetDateTime::UNIX_EPOCH + time::Duration::days(expire));
-
-                    Some((
-                        name,
-                        Arc::new(Shadow {
-                            crypt_pw: password,
-                            aging_policy,
-                            expiration_date,
-                        }),
-                    ))
-                } else {
-                    // Invalid password, skip the account
-                    None
-                }
-            });
+                    Arc::new(Shadow {
+                        crypt_pw: password,
+                        aging_policy,
+                        expiration_date,
+                    }),
+                ))
+            } else {
+                // Invalid password, skip the account
```

Comment:
Need to add a debug message here so folks can figure it out

Reply:
This is more common than you may think, it's normal to lock an account by replacing the pw hash with an x or ! so that they become locked. So I think this could be noisy. I'm adding the message, but only at a debug level.